packages/dd-trace/src/aiguard/sdk.js

```diff
@@ -0,0 +1,164 @@
+'use strict'
+
+const NoopAIGuard = require('./noop')
+const executeRequest = require('./client')
+const {
+  AI_GUARD_RESOURCE,
+  AI_GUARD_TARGET_TAG_KEY,
+  AI_GUARD_REASON_TAG_KEY,
+  AI_GUARD_ACTION_TAG_KEY,
+  AI_GUARD_BLOCKED_TAG_KEY,
+  AI_GUARD_META_STRUCT_KEY,
+  AI_GUARD_TOOL_NAME_TAG_KEY
+} = require('./tags')
+const log = require('../log')
+
+const ALLOW = 'ALLOW'
+
+class AIGuardAbortError extends Error {
+  constructor (reason) {
+    super(reason)
+    this.name = 'AIGuardAbortError'
+    this.reason = reason
+  }
+}
+
+class AIGuardClientError extends Error {
+  constructor (message, opts = {}) {
+    super(message)
+    this.name = 'AIGuardClientError'
+    if (opts.errors) {
+      this.errors = opts.errors
+    }
+    if (opts.cause) {
+      this.cause = opts.cause
+    }
+  }
+}
+
+class AIGuard extends NoopAIGuard {
+  constructor (tracer, config) {
+    super()
+
+    if (!config.apiKey || !config.appKey) {
+      log.error('AIGuard: missing api and/or app keys, use env DD_API_KEY and DD_APP_KEY')
+      this._initialized = false
+      return
+    }
+    this._tracer = tracer
+    this._headers = {
+      'DD-API-KEY': config.apiKey,
+      'DD-APPLICATION-KEY': config.appKey,
+    }
+    const endpoint = config.experimental.aiguard.endpoint || `https://app.${config.site}/api/v2/ai-guard`
+    this._evaluateUrl = `${endpoint}/evaluate`
+    this._timeout = config.experimental.aiguard.timeout
+    this._maxMessagesLength = config.experimental.aiguard.maxMessagesLength
+    this._maxContentSize = config.experimental.aiguard.maxContentSize
+    this._meta = { service: config.service, env: config.env }
+    this._initialized = true
+  }
+
+  #truncate (messages) {
+    const size = Math.min(messages.length, this._maxMessagesLength)
+    const result = messages.slice(-size)
```

Comment:
does it also make sense to grab any system message(s) as part of this for extra context? not sure what was done for python/what is expected for the `evaluate` API in general but maybe makes sense to ensure this is grabbed regardless of if `maxMessagesLength` is set?